src/Raven.Server/Smuggler/Documents/SmugglerExporter.cs

```diff
@@ -52,18 +59,44 @@ public ExportResult Export(DocumentsOperationContext context, Stream destination
 
                 if (OperateOnTypes.HasFlag(DatabaseItemType.Documents))
                 {
+                    progress.Progress = "Exporting Documents";
+                    onProgress?.Invoke(progress);
                     writer.WriteComma();
                     writer.WritePropertyName(("Docs"));
-                    var documents = DocumentsLimit.HasValue
-                        ? _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0, 0, DocumentsLimit.Value)
-                        : _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0);
+
+                    var batchSize = DocumentsLimit.HasValue ? Math.Min(Options.BatchSize, DocumentsLimit.Value) : Options.BatchSize;
```

Comment:
No, we had to do that to avoid problems in the way we were using the smuggler, makes no sense to do so now